### PR TITLE
Adding the swipe gesture

### PR DESCRIPTION
**Pull request by lpascal-ledger**
- Adding proxy `finger_swipe` methods in the backends, only really implemented in Speculos for now (need >= v0.9.0)
   - GUI-specific implementations still need to be done (in another PR?)
- Added an `Center(Element)` (position-only class, not tied to a layout or a use case).
  The thing is, a "swipe" is a bit like a "touch", it is not attached to a specific layout or use case. Still, I feel like basic swipe should be usable from anywhere.
  So I added a basic swipe like the touch, but also 2 more specific instructions: `SWIPE_CENTER_TO_LEFT` and `SWIPE_CENTER_TO_RIGHT` which I hope will embed 90% (100%? :crossed_fingers: ) of current use cases.
- Changed the `UseCase.TX_REVIEW` scenario to use the swipe instead of the tap.
  - This may need some thinking, as this specific use case may be better having a "next" method rather than a "tap" one. And it would be our responsibility to implement either a swipe or a tap inside it, directly into the `firmare` module (it should be managed there afterall). But I don't had time to think of it more.

### Tested on 
- monero tests
 ```--- a/tests/monero_client/monero_cmd.py
+++ b/tests/monero_client/monero_cmd.py
@@ -229,7 +229,7 @@ class MoneroCmd(MoneroCryptoCmd):
             instructions = get_nano_review_instructions(1)
         else:
             instructions = [
-                NavIns(NavInsID.USE_CASE_REVIEW_TAP)
+                NavInsID.SWIPE_CENTER_TO_LEFT
             ]
```

- Ethereum tests - no changes as it is abstracted into the `UseCase.TX_REVIEW` scenario

### CODE REVIEW

**Review by cedelavergne-ledger — DISMISSED** [review]
Good for me, tested with app-ethereum

File: src/ragger/navigator/navigation_scenario.py [hunk 1]
```diff
@@ -31,7 +31,10 @@ def __init__(self, device: str, use_case: UseCase, approve: bool):
                 self.pattern = "^Confirm$"
 
             elif use_case == UseCase.TX_REVIEW:
-                self.navigation = NavInsID.USE_CASE_REVIEW_TAP
+                if device.startswith("stax"):
+                    self.navigation = NavInsID.USE_CASE_REVIEW_TAP
```
Comment (xchapron-ledger) [inline src/ragger/navigator/navigation_scenario.py:35]: No SWIPE for `ADDRESS_CONFIRMATION` above?

File: src/ragger/navigator/navigation_scenario.py [hunk 2]
```diff
@@ -31,7 +31,10 @@ def __init__(self, device: str, use_case: UseCase, approve: bool):
                 self.pattern = "^Confirm$"
 
             elif use_case == UseCase.TX_REVIEW:
-                self.navigation = NavInsID.USE_CASE_REVIEW_TAP
+                if device.startswith("stax"):
```
Comment (xchapron-ledger) [inline src/ragger/navigator/navigation_scenario.py:34]:
```suggestion
                if device == "stax":
```
using `startswith` should probably only used on `nano`...

Comment (lpascal-ledger) [reply src/ragger/navigator/navigation_scenario.py:34]: Moreover I'm not sure in the first place why we are using strings when there is an enum.

File: src/ragger/navigator/navigation_scenario.py (hunk 1 above)
Comment (lpascal-ledger) [reply src/ragger/navigator/navigation_scenario.py:35]: I'm not aware of all the cases where the SWIPE is mandatory; I get from your comment that it is for `ADDRESS_CONFIRMATION` so I'll add it.

Comment (xchapron-ledger) [reply src/ragger/navigator/navigation_scenario.py:35]: I'd say that swipe is the flex equivalent of tap on stax :)

Comment (lpascal-ledger) [reply src/ragger/navigator/navigation_scenario.py]: Okay, that will be simpler then!

**Review by xchapron-ledger — COMMENTED** [review]
I'm wondering if we should create a `NavInsID.CONTINUE_NAVIGATION`
which would map on:
- `NavInsID.USE_CASE_REVIEW_TAP` for stax
- `NavInsID.SWIPE_CENTER_TO_RIGHT` for flex
- `NavInsID.RIGHT_CLICK` for nano

But maybe this will make things too opaque?

**Comment by lpascal-ledger** [comment]
This PR highlighted a bug in Speculos which will be fixed by [this PR](https://github.com/LedgerHQ/speculos/pull/486).

**Comment by lpascal-ledger** [comment]
@xchapron-ledger that may be interesting yes, but as for the implementation that would require adding / updates in the navigators, scenarios and firmware positions. This PR seems good enough as it is, I'd prefer having additional features in another one.

**Review by cedelavergne-ledger — APPROVED** [review]

**Pull request merged**